### DISCUSSION TITLE
IConfiguration is internally registered as singleton/scoped/transient ?

### DISCUSSION THREAD

**Discussion by abhay-profile** [dotnet/aspnetcore#55310] [Q&A]
I have researched and found on articles IConfiguration is registered as singleton but when I change something in appsettings IConfiguration gives updated value without rebuilding application. So it has to be atleast scoped ? I am confused please anyone explain this behaviour.

**Comment by gfoidl** [comment]
Even when a object is a singleton it's properties can be updated. The configuration objects isn't immutable / readable in that sense, and can be updated. That's the behavior you observe.

See e.g. [JSON configuration provider](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-8.0#json-configuration-provider) with it's `reloadOnChange` setting.

Further have a look at [Options pattern in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-8.0) to see the various possibilities of using monitored settings, etc.
